serial/serial_resources/process_entry.py

```diff
@@ -0,0 +1,149 @@
+        self.id = process_entry_data["id"] 
+
+    def add_text(self, dataset_name, value, expected_value=None):
+        """
+        Add text data to a process entry
+
+        Args: 
+        - dataset_name: User facing name of the dataset
+        - value: The value to be submitted
+        - expected_value: Optional argument to pass an expected value. If the 
+        values are the same, it would be considered a passing test result
+        """
+        if serial.debug:
+            print(f"Adding text data: {dataset_name} with value: {value} and expected value: {expected_value}")
+        dataset = serial.Datasets.get(dataset_name, "PARAMETRIC_QUALITATIVE", self.process_id)
+        data = {"type": "TEXT", "dataset_id": dataset.dataset_id, "value": value}
+        return self.client.make_api_request(f"/processes/entries/{self.id}", "PUT", data=data)
+
+    def add_number(self, dataset_name, value, usl=None, lsl=None):
+        """
+        Add numerical data to a process entry
+
+        Args: 
+        - dataset_name: User facing name of the dataset
+        - value: The value to be submitted
+        - usl: Optional argument to override the dataset's upper spec limit
+        - lsl: Optional argument to override the dataset's lower spec limit
+        """
+        if serial.debug:
+            print(f"Adding numerical data: {dataset_name} with value: {value} and usl: {usl} & lsl: {lsl}")
+        dataset = serial.Datasets.get(dataset_name, "PARAMETRIC_QUANTITATIVE", self.process_id)
+        data = {"type": "NUMERICAL", "dataset_id": dataset.dataset_id, "value": value}
+        if usl:
+            data["usl"] = usl
+        if lsl:
+            data["lsl"] = lsl
+        return self.client.make_api_request(f"/processes/entries/{self.id}", "PUT", data=data)
+
+    def add_file(self, dataset_name, path, bucket_type, file_name=None):
+        """
+        Add file data to a process entry
+
+        Args: 
+        - dataset_name: User facing name of the dataset
+        - path: Path to the file on your file system
+        """
+        if serial.debug:
+            pass
+        if not file_name:
+            file_name = os.path.basename(path)
+        # if in the future we need to do a better guess, we can use python-magic to determine the mimetype
+        files = {'file': (file_name, open(path, 'rb'), mimetypes.guess_type(path))} 
+        storage_object = self.client.make_api_request("/files", "POST", files=files)
+        dataset_type = "FILE"
+        if bucket_type == "images":
+            dataset_type = "IMAGE"
+        dataset = serial.Datasets.get(dataset_name, dataset_type, self.process_id)
+        data = {"type": "FILE", "dataset_id": dataset.dataset_id, "bucket_type": bucket_type, "file_id": storage_object["name"], "file_name": file_name}
+        return self.client.make_api_request(f"/processes/entries/{self.id}", "PUT", data=data)
+    
+    def add_boolean(self, dataset_name, value, expected_value):
```

Comment:
One thing that's missing here is ability to add data to a PASSFAIL field - which is one I want to deprecate entirely with the database rewrite - have all Boolean data (with expected_values) exist in booleans (now checkboxes)